Allow subscribers to be detached from namespace per event (#40154)
rails/rails: activesupport/CHANGELOG.md

> *  `ActiveSupport::Subscriber#detach_from` updated to take an `events:` argument, allowing
>     subscribers to detach from specific methods in a namespace while still receiving events for others.

Reviewer: ```suggestion
*  `ActiveSupport::Subscriber#detach_from` now accepts an `events:` argument that allows
    subscribers to detach from specific methods in a namespace while still receiving events from others.
```
Does this change make sense?

Author: Sure does! Thanks for the suggestion ❤️

    ```ruby

>     class StatsSubscriber < ActiveSupport::Subscriber

Reviewer 2: Small nitpick, let's use 2 spaces to indent this codeblock

      attach_to :active_record
    
      def sql(event)
        Statsd.timing("sql.#{event.payload[:name]}", event.duration)
      end

>       detach_from: :active_record, events: [:sql]

Reviewer 3: ```suggestion
      detach_from :active_record, events: [:sql]
```

Author: 👍 Thank you!
